src/server/aws/batchDeleteHandler.ts

```diff
@@ -0,0 +1,180 @@
+
+  /**
+   * Class for deleting records in batches from the database,
+   * when a user deletes their account. Consumes messages from
+   * the BatchDelete SQS queue in a synchronous, blocking way;
+   * only picks up a new message when the previous one has been
+   * completed (or had error), after a delay. If not actively
+   * processing messages, polls the queue on a schedule to discover
+   * messages.
+   * Queue polling starts by deafult
+   * once the class is instantiated and continues
+   * on a schedule.
+   * @param emitter The EventEmitter used by the class for scheduling
+   * poll events
+   * @param pollOnInit whether to start polling when the class is
+   * instantiated, primarily for testing (default=true);
+   */
+  constructor(public readonly emitter: EventEmitter, pollOnInit = true) {
+    this.sqsClient = new SQSClient({
+      region: config.aws.region,
+      endpoint: config.aws.endpoint,
+      maxAttempts: 3,
+    });
+    emitter.on(
+      BatchDeleteHandler.eventName,
+      async () => await this.pollQueue()
+    );
+    // Start the polling by emitting an initial event
+    if (pollOnInit) {
+      emitter.emit(BatchDeleteHandler.eventName);
+    }
+  }
+
+  /**
+   * Delete a message that has been handled from the
+   * BatchDelete queue
+   * @param message SQS Message recieved from BatchDelete queue
+   */
+  async deleteMessage(message: Message) {
+    const deleteParams = {
+      QueueUrl: config.aws.sqs.annotationsDeleteQueue.url,
+      ReceiptHandle: message.ReceiptHandle,
+    };
+    try {
+      await this.sqsClient.send(new DeleteMessageCommand(deleteParams));
+    } catch (error) {
+      const errorMessage = 'Error deleting message from queue ';
+      console.error(errorMessage, error);
+      console.error(JSON.stringify(message));
+      Sentry.addBreadcrumb({ message: errorMessage, data: message });
+      Sentry.captureException(error);
+    }
+  }
+
+  /**
+   * Handle messages from the batchDelete queue. Calls
+   * AccountDeleteDataService and forwards any errors to
+   * Cloudwatch and Sentry.
+   * @param body the body of the SQS message in the BatchDelete queue
+   * @returns whether or not the message was successfully handled
+   * (underlying call to AccountDeleteDataService completed without error)
+   */
+  async handleMessage(body: BatchDeleteMessage): Promise<boolean> {
+    const traceId = body.traceId ?? nanoid();
+    const userId = body.userId.toString();
+    try {
+      await new HighlightsDataService({
+        userId: userId.toString(),
+        db: {
+          writeClient: writeClient(),
+          readClient: readClient(),
+        },
+        apiId: 'service', // unused but required for inheritance
+        isPremium: false, //setting default `false` - but it shouldn't matter for delete
+      })
+        .deleteByAnnotationIds(body.annotationIds)
+        .then(() => successCallback('Annotations', userId, traceId));
+    } catch (error) {
```

Comment:
only change in this file is call successCallback on success, and failCallBack on err.

annotationsIds are printed within failCallBack